pkg/common/util/validation.go

```diff
@@ -0,0 +1,92 @@
+package util
```

Comment:
While this is the exact validation we need for registration API, I think the validation steps we take elsewhere might be subtly different. For instance, the datastore plugin does SPIFFE ID validation but it performs only a subset of these checks (it does not check trust domain because we store SPIFFE ID for other trust domains too).

Also, the growth of the common/util package worries me a little bit, because these functions are all exported (providing a publicly consumable api), and I suspect that in time a lot of them will need to be moved to their own packages. It's my fault for introducing util package to begin with :), golang blog says "don't do it": https://blog.golang.org/package-names

How would you feel about moving this `validation.go` file to the registration handler package instead? We will probably want to provide the public with a general purpose validator that they can use (perhaps in `api/` directory), but I suspect that might be only a subset of this...